packages/core/src/components/checkbox/checkbox.scss

```diff
@@ -0,0 +1,71 @@
+@use '../../helpers';
+@use '../../mixins';
+
+@mixin Checkbox() {
+  $size: helpers.space(2.75);
+
+  .ods-checkbox {
+    &,
+    + .ods-input-indicator::before {
+      border-radius: helpers.border-radius('medium');
+    }
+
+    // "outline" transition
+    + .ods-input-indicator::before {
+      transition: 0.15s ease-out;
+      transition-property: background-color;
+    }
+
+    // "tick" for [checked] indication
+    &:checked + .ods-input-indicator::after {
+      $border-width: 3px;
+      $width: $size / 2;
+
+      border-bottom: $border-width solid helpers.color('content-on-action');
+      border-right: $border-width solid helpers.color('content-on-action');
+      height: $size;
+      margin: -($border-width / 2) ($width / 2) 0;
+      opacity: 0;
+      transform: rotate(45deg) scale(0.6);
+      width: $width;
+    }
+
+    // "line" for [indeterminate] indication
+    &:indeterminate + .ods-input-indicator::after {
+      $height: 3px;
+
+      background-color: helpers.color('content-on-action');
+      height: $height;
+      opacity: 0;
+      top: calc(50% - #{$height} / 2);
+      transform: scale(0.6);
+      width: $size;
+    }
+  }
+
+  // indicating [checked] or [indeterminate] input
+  .ods-checkbox:checked,
+  .ods-checkbox:indeterminate {
+    + .ods-input-indicator::after {
+      opacity: 1;
+    }
```

Comment:
The `scale` here makes the "tick" and the "line" look blurry.
Also, we could use `transform-origin` rather than `margin` as it's more precise for translation.
Finally, there's an extra rule for `opacity`, but not necessary since the selector is the same.

Here's my suggestion.

| | before | after |
| --- | --- | --- |
| indeterminate | ![image](https://user-images.githubusercontent.com/18623773/105343176-3333dd80-5bd9-11eb-9083-f489faad08c6.png) | ![image](https://user-images.githubusercontent.com/18623773/105343224-3fb83600-5bd9-11eb-85de-4fd18135d6f6.png) |
| checked | ![image](https://user-images.githubusercontent.com/18623773/105343303-5ced0480-5bd9-11eb-85ac-811a164bc6bf.png) | ![image](https://user-images.githubusercontent.com/18623773/105343339-670f0300-5bd9-11eb-92cf-f54ca5aace2e.png) |

```suggestion    
    // "tick" for [checked] indication
    &:checked + .ods-input-indicator::after {
      $stroke-width: 2px;
      $width: $size * 0.65; // similar to `scale(0.6)`, but 0.65 makes it a nicer 14px
      $height: $width / 2;

      border: 0 solid helpers.color('content-on-action');
      border-width: 0 0 $stroke-width $stroke-width;
      height: $height;
      transform: rotate(-45deg);
      transform-origin: 110% 20%; // rotates rectangle from dead center
      width: $width;
    }

    // "line" for [indeterminate] indication
    &:indeterminate + .ods-input-indicator::after {
      $stroke-width: 2px;
      $width: $size * 0.55; // similar to `scale(0.6)`, but 0.55 makes it a nicer 12px
      $horizontal: ($size - $width) / 2;
      $vertical: ($size - $stroke-width) / 2; // vertical padding divided by two

      background-color: helpers.color('content-on-action');
      height: $stroke-width;
      margin: $vertical $horizontal;
      width: $width;
    }
  }
  
  // indicating [checked] or [indeterminate] input
  .ods-checkbox:checked,
  .ods-checkbox:indeterminate {
```

Reply:
Thanks, applied [your suggestion but slitghly modified](https://github.com/onfido/castor/pull/78/commits/995f3a32e45a6d2c3ff0ea6704015585deac2ca3).

Reply:
Looks good, managed to get rid of magic numbers too 👍🏻